tests/templates/test_subroutines/test_adder.py

```diff
@@ -0,0 +1,212 @@
+                None,
+                None,
+            ),
+        ],
+    )
+    def test_operation_result_args_None(
+        self, k, wires, mod, work_wires
+    ):  # pylint: disable=too-many-arguments
+        """Test the correctness of the Adder template output."""
+        dev = qml.device("default.qubit", shots=1)
+
+        @qml.qnode(dev)
+        def circuit(m):
+            qml.BasisEmbedding(m, wires=wires)
+            qml.Adder(k, wires, mod, work_wires)
+            return qml.sample(wires=wires)
+
+        if mod is None:
+            max = 2 ** len(wires)
+        else:
+            max = mod
+        for m in range(max):
+            assert np.allclose(
+                sum(bit * (2**i) for i, bit in enumerate(reversed(circuit(m)))), (m + k) % max
+            )
+
+    @pytest.mark.parametrize(
+        ("k", "wires", "mod", "work_wires", "msg_match"),
+        [
+            (
+                3,
+                [0, 1, 2, 3, 4],
+                12,
+                [4, 5],
+                "None of the wires in work_wires should be included in wires.",
+            ),
+        ],
+    )
+    def test_wires_error(
+        self, k, wires, mod, work_wires, msg_match
+    ):  # pylint: disable=too-many-arguments
+        """Test an error is raised when some wire in work_wires is in wires"""
+        with pytest.raises(ValueError, match=msg_match):
+            qml.Adder(k, wires, mod, work_wires)
+
+    @pytest.mark.parametrize(
+        ("k", "wires", "mod", "work_wires"),
+        [
+            (
+                4,
+                [0, 1, 2, 3],
+                7,
+                [4, 5],
+            ),
+            (
+                4,
+                [0, 1, 2, 3],
+                None,
+                [4, 5],
+            ),
+            (
+                4,
+                [0, 1, 2, 3],
+                8,
+                [4, 5],
+            ),
+        ],
+    )
+    def test_decomposition(self, k, wires, mod, work_wires):
+        """Test that compute_decomposition and decomposition work as expected."""
+
+        Adder_decomposition = qml.Adder(k, wires, mod, work_wires).compute_decomposition(
+            k, mod, work_wires, wires
+        )
+        op_list = []
+        # we perform m+k modulo mod
+        if mod == 2 ** (len(wires)):
+            qft_wires = wires
+        else:
+            qft_wires = work_wires[:1] + wires
+        # we perform m+k modulo mod
+        op_list.append(qml.QFT(qft_wires))
+        op_list.append(qml.PhaseAdder(k, wires, mod, work_wires))
+        op_list.append(qml.adjoint(qml.QFT)(qft_wires))
+
+        for op1, op2 in zip(Adder_decomposition, op_list):
+            qml.assert_equal(op1, op2)
+
+    @pytest.mark.jax
+    def test_jit_compatible(self):
+        """Test that the template is compatible with the JIT compiler."""
+
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+        m = 2
+        # m in binary
+        m_list = [0, 1, 0]
```

Comment:
you can pass "m" to BasisEmbedding 😄

Reply:
I know, but it raised an error when jitted, you can double check it yourself. So that's why I converted it into a list... This is an issue of BasisEmbedding not of our  Arithmetic templates.